interface/tuntap/TUNInterface_netbsd.c

```diff
@@ -0,0 +1,90 @@
+/* vim: set expandtab ts=4 sw=4: */
+/*
+ * You may redistribute this program and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "interface/tuntap/TUNInterface.h"
+#include "exception/Except.h"
+#include "interface/tuntap/BSDMessageTypeWrapper.h"
+#include "util/AddrTools.h"
+#include "util/events/Pipe.h"
+
+#include <errno.h>
+#include <ctype.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <stdlib.h>
+#include <stddef.h>
+#include <net/if.h>
+#include <string.h>
+#include <netdb.h>
+#include <net/if_tun.h>
+#include <netinet/in.h>
+#include <netinet6/in6_var.h>
+#include <netinet6/nd6.h>
+
+/* Tun Configurator for OpenBSD. */
+
+struct Iface* TUNInterface_new(const char* interfaceName,
+                                   char assignedInterfaceName[TUNInterface_IFNAMSIZ],
+                                   int isTapMode,
+                                   struct EventBase* base,
+                                   struct Log* logger,
+                                   struct Except* eh,
+                                   struct Allocator* alloc)
+{
+    if (isTapMode) { Except_throw(eh, "tap mode not supported on this platform"); }
+
+    int i;
+    // to store the tunnel device index
+    int ppa = 9;
+    // Open the descriptor
+    int tunFd = open("/dev/tun9", O_RDWR);
```

Comment:
The correct way is to loop over tun devices until you find one which opens properly. This is how we open the BPF device on Darwin: https://github.com/cjdelisle/cjdns/blob/master/interface/ETHInterface_darwin.c#L230